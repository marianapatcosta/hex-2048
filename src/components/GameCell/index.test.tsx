import React from 'react'
import { render, getByText } from '@testing-library/react'
import { GameCell } from './'

describe('GameCell', () => {
  const defaultProps = {
    x: 0,
    y: 0,
    z: 0,
  }
  test('should render without errors and match snapshot', () => {
    const {
      container: { firstChild },
    } = render(<GameCell {...defaultProps} value={2} />)

    expect(firstChild).toMatchSnapshot()
  })

  test('should display number 2 when value is 2', () => {
    const { container } = render(<GameCell {...defaultProps} value={2} />)

    getByText(container, 2)
  })

  test('should not display any number when value is 0', () => {
    const { queryByText } = render(<GameCell {...defaultProps} value={0} />)

    expect(queryByText(0)).not.toBeInTheDocument()
  })
})
