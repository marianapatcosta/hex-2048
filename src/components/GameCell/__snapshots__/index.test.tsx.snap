// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCell should render without errors and match snapshot 1`] = `
<div
  class="game-cell"
  data-value="2"
  data-x="0"
  data-y="0"
  data-z="0"
>
  <div
    class="game-cell game-cell--inner"
    data-value="2"
    data-x="0"
    data-y="0"
    data-z="0"
  >
    <span>
      2
    </span>
  </div>
</div>
`;
